lib/index.js

```diff
@@ -1723,38 +1749,35 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var seen = {};  // prevents testing overridden properties multiple times.
```

Comment:
Cool. Fixed in https://github.com/caolan/highland/commit/73a8cfe6f2a1fcb7de323ff1546ca126194edbfb, though I don't have IE8 lying around, so I can't confirm.